packages/vaex-hdf5/vaex/hdf5/export.py

```diff
@@ -156,13 +158,17 @@ def export_hdf5(dataset, path, column_names=None, byteorder="=", shuffle=False,
                 #     data_shape = column.bytes.shape
                 #     indices_shape = column.indices.shape
                 # else:
-
-                byte_length = dataset[column_name].str.byte_length().sum(selection=selection)
+                
+                byte_length = dataset[column_name].str.byte_length()
+                gc.collect()
+                byte_length = byte_length.sum(selection=selection)
+                gc.collect()
```

Comment:
```suggestion
                byte_length = dataset[column_name].str.byte_length().sum(selection=selection)
                gc.collect()
```
i think this should be enough, calling `byte_length()` will only build an expression, not use any memory.

Reply:
Sure! I'm happy to help!

From my debug, the main reasons why this is happening, are mainly 2 factors:

a) on "packages/vaex-core/vaex/__init__.py" there is a for loop that continuously increases memory usage, because of the "export_hdf5" method, used on this line: **vaex_df.export_hdf5(filename_hdf5, shuffle=False)** > line 545

    For this reason I made a call to gc.collect() at the end of each loop, but also one before the start of the loop, to keep memory as low as possible at that point.

    However, this was ok whilst looping through the chunks, but after the loop on init.pyt, another issue happened when trying to put together all the chunks back into a single file, which I will explain on point b)

b) This line of code:
     dfs = [vaex.file.open(p) for p in converted_paths]

     is implicitly calling the "export_hdf5" method in "packages/vaex-hdf5/vaex/hdf5/export.py" 

     Here, I've split the line of code you refer, because it help to keep the memory low (this 2 methods generate loads of garbage together, and when those are split, it helps to avoid the process from being killed) :

                byte_length = dataset[column_name].str.byte_length()
                gc.collect()
                byte_length = byte_length.sum(selection=selection)
                gc.collect()
                
      The other gc.collect() were located in other critical places were I found there was more memory consumption.

This was the only way I found to process a 4,6GB dataset (without reducing the chunksize), on a virtual machine with the following config:
        Ubuntu 20.04 LTS
        Python 3.8
        4 GB RAM
        no swap

So this dataset is only 15% over the RAM size (although the system uses already 1.4GB on its own)

The maximum recommended chunksize is 1/8 of the available RAM (example: 4GB RAM means chunksize = 4_000_000 / 8 = 500_000), as I tried to keep this chunksize as high as possible, to allow more flexibility.

I have reproduced the issue also on a EC2 with the configuration mentioned on the issue raised, but this config was less tolerant for my debug purpose.

Please let me know your thoughts on this and I will proceed accordingly.

Thanks!

Reply:
I will certainly get rid of the white-spaces, sorry about that! 👍